doc/how_to_guides/parallel_planning/src/parallel_planning_main.cpp

```diff
@@ -0,0 +1,300 @@
+                                                     const planning_interface::MotionPlanResponse& solution_b) {
+                                                    // If both solutions were successful, check which path is shorter
+                                                    if (solution_a && solution_b)
+                                                    {
+                                                      return robot_trajectory::path_length(*solution_a.trajectory_) <
+                                                             robot_trajectory::path_length(*solution_b.trajectory_);
+                                                    }
+                                                    // If only solution a is successful, return a
+                                                    else if (solution_a)
+                                                    {
+                                                      return true;
+                                                    }
+                                                    // Else return solution b, either because it is successful or not
+                                                    return false;
+                                                  });
+  RCLCPP_INFO(LOGGER, "'%s' chosen as best solution (Shortest path)", shortest_solution->planner_id_.c_str());
+  RCLCPP_INFO(LOGGER, "#####################################################");
+  return *shortest_solution;
+}
+
+/// \brief Utility class to create and interact with the parallel planning demo
+class Demo
+{
+public:
+  /// @brief
+  /// @param node
+  Demo(rclcpp::Node::SharedPtr node)
+    : node_{ node }
+    , moveit_cpp_{ std::make_shared<moveit_cpp::MoveItCpp>(node) }
+    , planning_component_{ std::make_shared<moveit_cpp::PlanningComponent>(PLANNING_GROUP, moveit_cpp_) }
+    , visual_tools_(node, "panda_link0", "parallel_planning_example", moveit_cpp_->getPlanningSceneMonitor())
+  {
+    moveit_cpp_->getPlanningSceneMonitor()->providePlanningSceneService();
+
+    visual_tools_.deleteAllMarkers();
+    visual_tools_.loadRemoteControl();
+
+    Eigen::Isometry3d text_pose = Eigen::Isometry3d::Identity();
+    text_pose.translation().z() = 1.75;
+    visual_tools_.publishText(text_pose, "Parallel Planning Tutorial", rvt::WHITE, rvt::XLARGE);
+    visual_tools_.trigger();
+  }
+
+  bool loadPlanningSceneAndQuery()
+  {
+    std::string hostname = "";
+    int port = 0.0;
+    std::string scene_name = "";
+
+    node_->get_parameter_or(std::string("warehouse.host"), hostname, std::string("127.0.0.1"));
+    node_->get_parameter_or(std::string("warehouse.port"), port, 33829);
+
+    if (!node_->get_parameter("warehouse.scene_name", scene_name))
+    {
+      RCLCPP_WARN(LOGGER, "Warehouse scene_name NOT specified");
+    }
+
+    moveit_warehouse::PlanningSceneStorage* planning_scene_storage = nullptr;
+
+    // Init DB connection
```

Comment:
Would recommend full words

```suggestion
    // Initialize database connection
```

bonus points if you say why that database is needed, e.g., "to load collision point clouds for the planning scene" or whatever